code/messaging/src/main/java/com/adobe/marketing/mobile/aepcomposeui/LargeImageUI.kt

```diff
@@ -0,0 +1,56 @@
+/*
+  Copyright 2025 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.aepcomposeui
+
+import androidx.compose.runtime.mutableStateOf
+import com.adobe.marketing.mobile.aepcomposeui.state.LargeImageCardUIState
+import com.adobe.marketing.mobile.aepcomposeui.uimodels.LargeImageTemplate
+
+/**
+ * Implementation of the [AepUI] interface used in rendering a UI for a [LargeImageTemplate].
+ *
+ * @property template The template associated with the large image UI.
+ * @property _state The current state of the large image UI.
+ */
+class LargeImageUI(
+    private val template: LargeImageTemplate,
+    state: LargeImageCardUIState
+) : AepUI<LargeImageTemplate, LargeImageCardUIState> {
```

Comment:
Let's make a base class that does the boilerplate template and state handling so that all of your implementations can inherit from this.

```
class LargeImageUI(
    private val template: LargeImageTemplate,
    state: LargeImageCardUIState
) : BaseAepUI<LargeImageTemplate, LargeImageCardUIState>(template, state)


abstract class BaseAepUI<T : AepUITemplate, S : AepCardUIState>(
    private val template: T,
    state: S
) : AepUI<T, S> {
    private val _state = mutableStateOf(state)

    override fun updateState(newState: S) {
        _state.value = newState
    }
    
    override fun getTemplate(): T {
        return template
    }

    override fun getState(): S {
        return _state.value
    }
}
```

Reply:
+1